### DISCUSSION TITLE
Compilation error in htable.f90 with ifort

### DISCUSSION THREAD

**Discussion by timofeymukha** [ExtremeFLOW/neko#336] [General]
Hello!

Tried to compile on tetralith with `ifort 18.0.1 20171018` and got the following error during `make install`

```
mpif90  -g -c -o adt/htable.o adt/htable.f90
adt/htable.f90(518): error #7137: Any procedure referenced in a PURE procedure, including one referenced via a defined operation or assignment, must have an explicit interface and be declared PURE.   [ISO_C_BINDING^C_ASSOCIATED_PTR]
          res = c_associated(kp%ptr, key%ptr)
----------------^
compilation aborted for adt/htable.f90 (code 1)
```

Something expected, or is a fix needed?

**Comment by njansson** [comment]
Indeed expected for that version of ifort and its interpretation of the standard wrt `pure` in the  iso_c_interface

Either use a more recent version of ifort (version 19 or newer), or (unless you need accelerator support) simply change that line to `res = .false.`

**Comment by njansson** [comment]
PS. Moved this to the new discussion board instead
